moto/events/models.py

```diff
@@ -224,6 +224,9 @@ def create_from_cloudformation_json(
         properties = cloudformation_json["Properties"]
         properties.setdefault("EventBusName", "default")
 
+        if "EventPattern" in properties:
+            properties["EventPattern"] = json.dumps(properties["EventPattern"])
```

Comment:
The other way around. The value of `properties["EventPattern"]` is a `dict`:

```

s = {'detail': {'adeId': ['acNdVxeLLewgrwuR'], 'serviceName': ['EXZvZQfDRUnSRCfX']}, 'detail-type': ['meteor']}, cls = None, object_hook = None, parse_float = None, parse_int = None, parse_constant = None, object_pairs_hook = None, kw = {}

 ...(truncated)...

        if isinstance(s, str):
            if s.startswith('\ufeff'):
                raise JSONDecodeError("Unexpected UTF-8 BOM (decode using utf-8-sig)",
                                      s, 0)
        else:
            if not isinstance(s, (bytes, bytearray)):
>               raise TypeError(f'the JSON object must be str, bytes or bytearray, '
                                f'not {s.__class__.__name__}')
E               TypeError: the JSON object must be str, bytes or bytearray, not dict
```

`json.dumps(...)` converts it to a string representing a json dictionary.

The error is misleading. It assumes all templates are JSON. They are not.

Reply:
Makes sense, I was confused by the [test](https://github.com/EBSCOIS/moto/pull/1/files#diff-a94a68a60516981adcf82ffb3c957859a70fcad018bc390faa0af1e0e0dc6220R48) because I thought it was a mixing of json and yaml.